zebra-chain/src/transaction/serialize.rs

```diff
@@ -309,6 +357,12 @@ impl ZcashDeserialize for Transaction {
                 let expiry_height = block::Height(reader.read_u32::<LittleEndian>()?);
                 let inputs = Vec::zcash_deserialize(&mut reader)?;
                 let outputs = Vec::zcash_deserialize(&mut reader)?;
+                let shielded_spends = Vec::zcash_deserialize(&mut reader)?;
+                let shielded_outputs = Vec::zcash_deserialize(&mut reader)?;
+                let shielded_data =
+                    deserialize_shielded_data(&mut reader, shielded_spends, shielded_outputs)?;
+                let sapling_value_balance = (&mut reader).zcash_deserialize_into()?;
```

Comment:
If inputs and outputs are `None`, these fields do not exist in `V5`:
https://github.com/ZcashFoundation/zebra/issues/1829#issuecomment-800133051

I will add a test for this bug to the design in #1863.

After #1863 is completed, please revise all the code in this PR, based on the latest spec.

<img width="840" alt="Screen Shot 2021-03-17 at 11 07 55" src="https://user-images.githubusercontent.com/8951843/111399421-13202680-8711-11eb-9646-ee16b7eef71b.png">